exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/AliasesTests.kt

```diff
@@ -232,4 +232,28 @@ class AliasesTests : DatabaseTestsBase() {
             assertEquals(foreignKey, aliasForeignKey)
         }
     }
+
+    @Test
+    fun testAccessDataByQueryExpressionAlias() {
+        val stables = object : UUIDTable("Stables") {}
+
+        val facilities = object : UUIDTable("Facilities") {
+            val stableId = reference("stable_id", stables)
+        }
+
+        withTables(stables, facilities) {
+            val countExpression = facilities.id.count()
+            val countSubquery = facilities
+                .select(countExpression)
+                .where { facilities.stableId eq stables.id }
+                .expression(countExpression)
+                .alias("count_query")
```

Comment:
```kt
val countSubquery = wrapAsExpression<Long>(
        facilities
            .select(countExpression)
            .where { facilities.stableId eq stables.id }
).alias("count_query")
```

Reply:
Thank you, I haven't noticed that function... In this variant the type should be defined by user, it looks simpler (in my variant it was inferred from the provided column type of the expression).

I added here one more extension function `fun <T : Any> AbstractQuery<*>.asExpression() = wrapAsExpression<T>(this)`. It could be useful to call it directly on the column to get an expression.